.github/workflows/env/action.yml

```diff
@@ -5,11 +5,34 @@ runs:
   steps:
     - name: Install dependencies
       shell: bash
-      run: sudo apt-get install -y llvm clang dwz cmake curl unzip
-    - name: Set up Go ${{matrix.go}}
+      run: |
+        # Ubuntu has ARM64 packages in an entirely different repo, so we can't just
+        # `dpkg --add-architecture arm64` here: we have to add a custom sources.list first.
+        sudo tee /etc/apt/sources.list.d/ubuntu.sources <<EOF > /dev/null
+        Types: deb
+        URIs: http://azure.archive.ubuntu.com/ubuntu/
+        Suites: noble noble-updates noble-backports
+        Components: main universe restricted multiverse
+        Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
+        Architectures: amd64
+
+        Types: deb
+        URIs: http://azure.ports.ubuntu.com/ubuntu-ports/
+        Suites: noble noble-updates noble-backports
+        Components: main universe restricted multiverse
+        Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg
+        Architectures: arm64
+        EOF
+        
+        sudo dpkg --add-architecture arm64
+        sudo apt-get update -y
+        
+        sudo apt-get install -y llvm clang dwz curl unzip gcc-aarch64-linux-gnu \
+          libc6-arm64-cross qemu-user-binfmt libc6:arm64
+    - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: ${{matrix.go}}
+        go-version: stable
```

Comment:
Nit: Using `stable` as version for Go comes with the risk of breaking CI if there is an unexpected change in Go. E.g. there could be changes similar to https://github.com/golang/go/issues/67401, that could break assumptions.
Manually updating the Go version allows to make sure CI does not break on events, that are happening outside of the scope and control of this repo/maintainership.

Ideally, this action should receive some version information as argument. Using `$matrix.go` here assumes that in the executed job `matrix.go` is defined and makes it hard to follow/review.

Reply:
> Nit: Using stable as version for Go comes with the risk of breaking CI if there is an unexpected change in Go. E.g. there could be changes similar to https://github.com/golang/go/issues/67401, that could break assumptions.

I'm merely mirroring the previous logic here, but I'd also be OK with hardcoding a particular Go version globally. IMO we should just have one global Go version for all CI workflows and adding it as an argument would force us to duplicate the version everywhere.

> Using $matrix.go here assumes that in the executed job matrix.go is defined and makes it hard to follow/review.

Yeah, that's why I opted to just hardcode it directly in the action. Also, all callers either passed "stable" already or forgot to set it entirely.